desktop/cmp/persistenceManager/PersistenceManager.ts

```diff
@@ -0,0 +1,152 @@
+                    items: [
+                        popover({
+                            omit: omitDefaultMenuComponent,
+                            item: button({
+                                text: model.getHierarchyDisplayName(selectedView?.name) ?? `-`,
+                                icon: isShared ? Icon.users() : Icon.bookmark(),
+                                rightIcon: Icon.chevronDown(),
+                                outlined: true
+                            }),
+                            content: div({
+                                items: [
+                                    div({
+                                        className: 'xh-popup__title',
+                                        item: capitalize(pluralize(displayName))
+                                    }),
+                                    objMenu({minimal})
+                                ]
+                            }),
+                            placement: 'bottom-start'
+                        }),
+                        saveButton()
+                    ]
+                }),
+                manageDialog({omit: !manageDialogModel}),
+                saveDialog({omit: !saveDialogModel})
+            );
+        }
+    });
+
+//------------------------
+// Implementation
+//------------------------
+
+const saveButton = hoistCmp.factory<PersistenceManagerModel>({
+    render({model}) {
+        return button({
+            icon: Icon.save(),
+            tooltip: `Save changes to this ${model.entity.displayName}`,
+            intent: 'primary',
+            omit: !model.enableTopLevelSaveButton || !model.canSave,
+            onClick: () => model.saveAsync(false).linkTo(model.loadModel)
+        });
+    }
+});
+
+const objMenu = hoistCmp.factory<PersistenceManagerProps>({
+    render({model, minimal}) {
+        const {loadModel, entity} = model,
+            items = [];
+
+        model.viewTree.forEach(it => {
+            items.push(buildView(it, model));
+        });
+
+        if (minimal) return menu({items});
+        return menu({
+            items: [
+                ...items,
+                menuDivider(),
+                menuItem({
+                    icon: Icon.save(),
+                    text: 'Save',
+                    disabled: !model.canSave,
+                    onClick: () => model.saveAsync(false).linkTo(loadModel)
+                }),
+                menuItem({
+                    icon: Icon.copy(),
+                    text: 'Save as...',
+                    onClick: () => model.saveAsAsync().linkTo(loadModel)
+                }),
+                menuItem({
+                    icon: Icon.reset(),
+                    text: 'Revert View',
+                    disabled: !model.isDirty,
+                    onClick: () => model.resetAsync().linkTo(loadModel)
+                }),
+                menuItem({
+                    icon: Icon.refresh(),
+                    text: 'Reset Defaults',
```

Comment:
Our `Grid` context menu item text is `Restore Grid Defaults`. Maybe this should be `Restore Default View`?

Reply:
Or do we call this `Clear View` or something that more closely represents the implementation? I'm not sure what's best here.  In any case, I think we'll want a prop to enable/disable this option since I could imagine there being apps that _always_ want the user to be in _some_ view.